### PR TITLE
Animations with fluidsim-core

### LINKED ISSUE #118: sim.output.phys_fields.animate function to create movies and GIF
I think it's a very nice feature of core fluidsim solvers to have this function. Users can very easily create simple 2d animations representing the flow simulated. It's very useful to "see what happens" and of course for communication (BTW, we don't use it enough).

Animations in fluidsim are based on this method: https://fluidsim.readthedocs.io/en/latest/generated/fluidsim.base.output.movies.html#fluidsim.base.output.movies.MoviesBase.animate

It would be very nice to have something similar for Snek. I don't know if it could be based  on the code of `fluidsim.base.output.movies.MoviesBase`. If it could, we would have to move this class in `fluidsim_core` (super easy).

I don't include this issue in our 0.8.0 milestone, but it would be very useful for #92 since it would be very easy to create nice animated GIF that could be included in the README and in the doc. Such GIF would also be useful for communication on social media (see for example https://twitter.com/me3250).

### PR DESCRIPTION
**Pull request by paugier**
Fixes #118 and #115 .

This is still a work in progress and needs https://foss.heptapod.net/fluiddyn/fluidsim/-/merge_requests/317 (not yet merged).

At this stage we already can do:

```
from fluidsim import load 
# or from snek5000 import load
sim = load()
sim.output.phys_fields.animate(tmax=400, dt_frame_in_sec=0.1, dt_equations=5, interactive=True)
# or sim.output.phys_fields.animate(tmax=400, dt_frame_in_sec=0.1, dt_equations=5, save=True)
```

It needs to be improved in particular for:

- get the times associated with the files without loading the whole files
- use a regular grid for vectors (useful to skip some vectors)
- support `set_equation_crosssession` (for 3d)
- clean up all the quick and dirty code
- read only what is necessary for the movie and not all the file
  * only the necessary field(s)
  * only the necessary cross-section

I guess we will need to add some features in pymech (?)

### CODE REVIEW

File: src/snek5000/output/phys_fields.py [hunk 1]
```diff
@@ -140,3 +163,151 @@ def change_reader(self, reader_key):
...
+                    vx_quiver.append(elem.vel[0, iz, iy, ix])
+                    vy_quiver.append(elem.vel[1, iz, iy, ix])
+
+        ax.quiver(x_quiver, y_quiver, vx_quiver, vy_quiver)
+
+    @staticmethod
+    def time_from_path(path):
+        with open(path, "rb") as file:
+            header = read_header(file)
+        return header.time
```
Comment (ashwinvis) [inline src/snek5000/output/phys_fields.py:298]: `return read_header(path).time` should do it. I made a recent change which accepts both a file object and a path.

**Comment by paugier** [comment]
We have some issues with the TGV simulation. 

1. The name of the directory is wrong ("tgv_run_6x6x6_V1pix1pix1pi" while `sim.params.oper.Lx = 2 *pi`)
2. I can't use adaptative dt
3. plot is quite wrong (I think related to the first problem)
4. `params.nek.general._print_docs()` nearly empty...

@ashwinvis @akhoubani do you understand better than me?

**Comment by akhoubani** [comment]
> @akhoubani do you understand better than me?

I'll check and let you know.

**Comment by paugier** [comment]
How did you get that? Did you execute the script in the PR?

What should I do to really get dt adapted? With this script the time step is just fixed by the parameter `dt`.

**Comment by akhoubani** [comment]
> How did you get that? Did you execute the script in the PR?

Yes, I executed the script in the PR.

> 
> What should I do to really get dt adapted? With this script the time step is just fixed by the parameter `dt`.

Comment targetCFL [here](https://github.com/snek5000/snek5000/blob/main/docs/examples/snek5000-tgv/src/snek5000_tgv/tgv.par.cfg) and also in the script. The default value of CFL is 0.5. Now you can change dt and it will be adapted as far as I know.

**Comment by akhoubani** [comment]
@paugier If `dt` is larger than `0`, the `CFL` determines the upper bound for `dt`. Nek5000 will choose the minimum of `dt` and `dt_CFL`. For the script in the PR, `dt` is always smaller compared to `dt_CFL`. Therefore it will not be changed.

**Comment by akhoubani** [comment]
`sim.output.phys_fields.plot_hexa()` gives a plot with velocity vectors only for values of `x` and `y` larger than `0`. It excludes the negative values of `x` and `y`.

**Comment by paugier** [comment]
@ashwinvis Please can you have a look at it and potentially merge!

@akhoubani Please can you try these new features on your data?

**Pull request merged**